packages/wonder-blocks-grid/components/cell.js

```diff
@@ -104,61 +98,70 @@ class Cell extends React.Component<Props> {
-        // add back in the missing gutter widths:
-        // (gutterWidth * (cols - 1)). This gives us to full width of
-        // this particular cell.
-        const calcWidth = `calc(${contentWidth} * ${cols /
-            totalColumns} + ${gutterWidth * (cols - 1)}px)`;
-
-        let contents = children;
-
-        // If the contents are a function then we call it with the mediaSize,
-        // totalColumns, and cols properties and render the return value.
-        if (typeof contents === "function") {
-            contents = contents({mediaSize, totalColumns, cols});
-        }
-
-        // Render a fixed-width cell (flex-basis: size, flex-shrink: 0)
-        // that matches the intended width of the cell
         return (
-            <FixedWidthCell width={calcWidth} style={style}>
-                {contents}
-            </FixedWidthCell>
+            <MediaLayout>
+                {({mediaSize, mediaSpec}) => {
+                    // Get the settings for this particular size of grid
+                    const {totalColumns, gutterWidth, marginWidth} = mediaSpec[
+                        mediaSize
+                    ];
+
+                    const cols = Cell.getCols(this.props, mediaSize);
+
+                    // If no columns are specified then we just don't render this cell
+                    if (cols === undefined || cols === null || cols === 0) {
+                        return null;
+                    }
+
+                    if (cols > totalColumns) {
+                        throw new Error(
+                            `Specified columns ${cols} is greater than the maximum ` +
+                                `${totalColumns} at the ${mediaSize} grid size.`,
+                        );
+                    }
+
+                    // We need to start by calculating the total width of all the "content"
+                    // We do this by starting with the full width (100%) and then
+                    // subtracting all of the gutter spaces inbetween the cells
+                    // (gutterWidth * (totalColumns - 1)) and the width of the two margins
+                    // (marginWidth * 2).
+                    const contentWidth = `(100% - ${gutterWidth *
+                        (totalColumns - 1)}px - ${marginWidth * 2}px)`;
+
+                    // Now that we have the full width we can calculate the width of this
+                    // particular cell by multiplying the full width (allCellWidth) by
+                    // the ratio of this cell (cols / totalColumns). But we then need to
+                    // add back in the missing gutter widths:
+                    // (gutterWidth * (cols - 1)). This gives us to full width of
+                    // this particular cell.
+                    const calcWidth = `calc(${contentWidth} * ${cols /
+                        totalColumns} + ${gutterWidth * (cols - 1)}px)`;
+
+                    let contents = children;
```

Comment:
I don't like `let` in these situations. Usually indicates the need for a method.

What do you think about a `getChildren` method that can then return different values based on this state? That way this becomes `const`.